src/osd/osd_types.cc

```diff
@@ -1943,6 +1945,8 @@ void object_stat_sum_t::decode(bufferlist::iterator& bl)
 #endif
   if (!decode_finish) {
     ::decode(num_bytes, bl);
+    if (struct_v >= 17)
+      ::decode(num_bytes_overlap, bl);
```

Comment:
Also, in order to introduce this we should really have a way to determine whether the value is accurate.  I don't think we can sanely initialize this to the right value in the 'else' compat decode case (which makes sense; if we could we wouldn't need to add this field).  Probably we'd need to pick a "safe" upper or lower bound and then set a flag so that it is silently corrected during the next scrub.